### PR TITLE
x86_64: correctly detect mode32 for atomic_store

### PR DESCRIPTION
**Pull request by carenas**
test 92 case 1 not failing because of an unlucky choice of test values.

### CODE REVIEW

File: sljit_src/sljitNativeX86_common.c [hunk 1]
```diff
@@ -3549,6 +3549,15 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_atomic_store(struct sljit_compiler
 	INC_SIZE(1);
 	*inst = GROUP_LOCK;
 
+#if (defined SLJIT_CONFIG_X86_64 && SLJIT_CONFIG_X86_64)
+	if (!compiler->mode32) {
+		inst = (sljit_u8*)ensure_buf(compiler, 1 + 1);
+		FAIL_IF(!inst);
+		INC_SIZE(1);
+		*inst = REX_W;
```
Comment (zherczeg) [inline sljit_src/sljitNativeX86_common.c:3557]: This prefix should be never added manually.

You can add a `op = GET_OPCODE(op);` after the SLJIT_ASSERT in the beginning of the function, and replace all `GET_OPCODE(op)` to `op`.

**Review by zherczeg — CHANGES_REQUESTED** [review]
Nice catch!

File: sljit_src/sljitNativeX86_common.c (hunk 1 above)
Comment (zherczeg) [reply sljit_src/sljitNativeX86_common.c:3558]: Why this is resolved? This code should be deleted.

Comment (zherczeg) [reply sljit_src/sljitNativeX86_common.c:3558]: To give background:
`compiler->mode32 = op != SLJIT_MOV && op != SLJIT_MOV_P;` does this, but I forgot that `op` has a flag, so it should be GET_OPCODE() guarded. Then, I realized we can do this early.

Comment (carenas) [reply sljit_src/sljitNativeX86_common.c]: My bad, didn't understood your comment about the prefix, and indeed the root cause of the bug, thanks or your explanation

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**